### ISSUE TITLE
Keyboard Shortcut Fails: Ctrl+W Word Deletion for Non-Alphanumeric Characters

### ISSUE THREAD

**Issue by devsunb** [anthropics/claude-code#2202]
**Bug Description**
Ctrl+W word deletion does not work with non-alphanumeric characters such as CJK.

**Environment Info**
- Platform: darwin
- Terminal: tmux
- Version: 1.0.25
- Feedback ID: 35811eb2-8a8a-4c4a-bf4f-3bf7aa49a739

**Comment by Kolger** [comment]
#1943
